app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id', dependent: :destroy
+  has_many :comments, foreign_key: 'author_id', dependent: :destroy
+  has_many :likes, foreign_key: 'author_id', dependent: :destroy
+
+  def recent_posts(limit = 3)
+    posts.order(created_at: :desc).limit(limit)
+  end
+
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
```

Comment:
- [ ] It would be nice to use one of the [Rails Active Callbacks (https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to `database`, then make the `update_posts_counter` method `private` 😁 
It can be similar to 👇 

```ruby
 after_save :update_posts_counter

  private

  def update_posts_counter
    author.increment!(:posts_counter)
  end
```